lib/Jsrt/Jsrt.cpp

```diff
@@ -156,6 +204,198 @@ CHAKRA_API JsCreateRuntime(_In_ JsRuntimeAttributes attributes, _In_opt_ JsThrea
+        VALIDATE_INCOMING_FUNCTION(function, scriptContext);
+
+        if(cargs == 0 || args == nullptr) {
+            return JsErrorInvalidArgument;
+        }
+
+        for(int index = 0; index < cargs; index++)
+        {
+            VALIDATE_INCOMING_REFERENCE(args[index], scriptContext);
+        }
+
+        Js::JavascriptFunction *jsFunction = Js::JavascriptFunction::FromVar(function);
+        Js::CallInfo callInfo(cargs);
+        Js::Arguments jsArgs(callInfo, reinterpret_cast<Js::Var *>(args));
+
+#if ENABLE_TTD
+        double ttdStartTime = -1.0;
+
+        if(PERFORM_JSRT_TTD_RECORD_ACTION_CHECK(scriptContext))
+        {
+            TTD::NSLogEvents::EventLogEntry* callEvent = scriptContext->GetThreadContext()->TTDLog->RecordJsRTCallFunction(scriptContext, scriptContext->TTDRootNestingCount, hostCallbackId, jsFunction, cargs, jsArgs.Values);
+            TTD::TTDRecordJsRTFunctionCallActionPopper actionPopper(scriptContext, callEvent);
+
+            if(scriptContext->TTDRootNestingCount == 0)
+            {
+                TTD::EventLog* elog = scriptContext->GetThreadContext()->TTDLog;
+                elog->ResetCallStackForTopLevelCall(elog->GetLastEventTime(), hostCallbackId);
+
+                ttdStartTime = elog->GetCurrentWallTime();
+            }
+
+            Js::Var varResult = jsFunction->CallRootFunction(jsArgs, scriptContext, true);
+            if(result != nullptr)
+            {
+                *result = varResult;
+                Assert(*result == nullptr || !Js::CrossSite::NeedMarshalVar(*result, scriptContext));
+            }
+
+            actionPopper.NormalReturn(result != nullptr ? *result : nullptr);
+        }
+        else
+        {
+            Js::Var varResult = jsFunction->CallRootFunction(jsArgs, scriptContext, true);
+            if(result != nullptr)
+            {
+                *result = varResult;
+                Assert(*result == nullptr || !Js::CrossSite::NeedMarshalVar(*result, scriptContext));
+            }
+        }
+
+        //Update the time elapsed since a snapshot if needed
+        if(ttdStartTime >= 0.0)
+        {
+            TTD::EventLog* elog = scriptContext->GetThreadContext()->TTDLog;
+
+            double ttdEndTime = elog->GetCurrentWallTime();
+            elog->IncrementElapsedSnapshotTime(ttdEndTime - ttdStartTime);
+        }
+#else
+        Js::Var varResult = jsFunction->CallRootFunction(jsArgs, scriptContext, true);
+        if(result != nullptr)
+        {
+            *result = varResult;
+            Assert(*result == nullptr || !Js::CrossSite::NeedMarshalVar(*result, scriptContext));
+        }
+#endif
+
+        return JsNoError;
+    });
+}
+
+/////////////////////
+
+CHAKRA_API JsCreateRuntime(_In_ JsRuntimeAttributes attributes, _In_opt_ JsThreadServiceCallback threadService, _Out_ JsRuntimeHandle *runtimeHandle)
+{
+    return CreateRuntimeCore(attributes, nullptr, nullptr, UINT_MAX, UINT_MAX, threadService, runtimeHandle);
```

Comment:
Please add comments on what are these values for. Similarly for CreateContextCore